evap/staff/importers.py

```diff
@@ -572,26 +572,57 @@ def write_enrollments_to_db(self, semester, vote_start_datetime, vote_end_date):
+
+        student_phrase = ngettext("{count} student", "{count} students", student_count).format(count=student_word)
+        responsible_phrase = ngettext("{count} responsible", "{count} responsibles", responsible_count).format(
+            count=responsible_word
+        )
+        evaluation_phrase = ngettext(
+            "{count} course/evaluation", "{count} courses/evaluations", evaluation_count
+        ).format(count=evaluation_word)
 
-        msg = format_html(
-            _("Successfully created {} courses/evaluations, {} participants and {} contributors:"),
-            len(self.evaluations),
-            len(students_created),
-            len(responsibles_created),
+        message = _("Successfully created {evaluation_phrase}, {student_phrase} and {responsible_phrase}").format(
+            responsible_phrase=responsible_phrase, student_phrase=student_phrase, evaluation_phrase=evaluation_phrase
         )
-        msg += create_user_list_html_string_for_message(students_created + responsibles_created)
-        self.success_messages.append(msg)
+        if student_count == 0 and responsible_count == 0 and evaluation_count == 0:
+            message = "Nothing changend"
+        elif student_count > 0 or responsible_count > 0:
+            message += ":"
+            message += create_user_list_html_string_for_message(students_created)
+            message += create_user_list_html_string_for_message(responsibles_created)
+        self.success_messages.append(message)
 
     def create_test_success_messages(self):
         filtered_users = [user_data for user_data in self.users.values() if not user_data.user_already_exists()]
 
         self.success_messages.append(_("The test run showed no errors. No data was imported yet."))
-        msg = format_html(
-            _("The import run will create {} courses/evaluations and {} users:"),
-            len(self.evaluations),
-            len(filtered_users),
+
+        user_count = len(filtered_users)
+        user_word = pgettext("count", "no") if user_count == 0 else str(user_count)
+        user_phrase = ngettext("{count} user", "{count} users", user_count).format(count=user_word)
+        evaluation_count = len(self.evaluations)
+        evaluation_word = pgettext("count", "no") if evaluation_count == 0 else str(evaluation_count)
+        evaluation_phrase = ngettext(
+            "{count} course/evaluation", "{count} courses/evaluations", evaluation_count
+        ).format(count=evaluation_word)
+
+        msg = _("The import run will create {evaluation_phrase} and {user_phrase}").format(
+            evaluation_phrase=evaluation_phrase, user_phrase=user_phrase
         )
-        msg += create_user_list_html_string_for_message(filtered_users)
+        if user_count == 0:
+            msg += format_html(_("."))
+        else:
+            msg += format_html(_(":"))
+            msg += create_user_list_html_string_for_message(filtered_users)
```

Comment:
Neither `.` nor `:` are translated or escaped, so both `_` and `format_html` are unneeded. 

Additionally, the `create_user_list_html_string_for_message` returns an already escaped string (it is an instance of `django.utils.safestring.SafeData`), but when you "add" it onto another string, this knowledge is lost and the created string will be escaped again - we don't want that. This is where `format_html` comes in, it does a `conditional_escape` on the inserted parameters (except the first one), so the safe data wont be escaped again. Since your phrases from above need to be escaped too, you probably want something like `format_html("{}: {}", msg, create_user_list(...))`

Interestingly, your tests don't fail although this gets escaped twice :thinking: